roles/smartgateway/tasks/events.yml

```diff
@@ -0,0 +1,72 @@
+---
+# tasks file for smartgateway
+- name: Write out configmap for smart gateway
+  template:
+    dest: /tmp/configmap.yaml
+    src: "{{ service_type }}-configmap.yaml.j2"
+
+- name: Smart gateway ConfigMap
+  k8s:
+    src: /tmp/configmap.yaml
+
+- name: Start the smart gateway
+  k8s:
+    resource_definition:
+      apiVersion: apps.openshift.io/v1
+      kind: DeploymentConfig
+      metadata:
+        name: '{{ meta.name }}-smartgateway'
+        namespace: '{{ meta.namespace }}'
+      spec:
+        replicas: '{{ size }}'
+        template:
+          metadata:
+            labels:
+              app: smartgateway_events_consumer
```

Comment:
Based on our discussions around labeling and the plan here: https://github.com/redhat-service-assurance/telemetry-framework/tree/csibbitt-594-fix-labels-and-servicemonitor/deploy#before

Suggest:
```
app: smart-gateway-events
```
and no other labels unless you find they are required for something.

The metrics consumer should also become `app: smart-gateway-metrics`

Reply:
So I figured out how to actually not even add the `events` or `metrics` to the label. The `ServiceMonitor` requires a few items, one of which is a `port` name. Instead of overloading the label (a smart gateway is a smart gateway?), I instead named the port via `{{ service_type }}` which sets the value there to `metrics` or `events`.

I also then wrapped the creation of the `ServiceMonitor` to only get created when creating a `metrics` SG. I think this approach might be a bit cleaner. We can also add a `smart-gateway-type` label if we really needed or wanted to, but I don't think it's going to be necessary.

Reply:
Is there some reason we don't want the events SG to get scraped by prometheus? Is it just empty? I figure we'd want to publish some stats on the # of events processed or something. If there's nothing useful there now then this is fine, otherwise I'm not convinced the ports should actually be named differently since they both refer to a prometheus scrape endpoint (I'd even argue for some standard name like 'prom-endpoint' to make it easier for prometheus to find them all without eventually having to write a servicemonitor for each).

Not blocking on this, but curious about the decision.

Reply:
Hrmmmmmmmmmm that's a very good point I hadn't thought of :)

In my head I had kind of separated out metrics smart gateways from events smart gateway due to what they connect to or provide. The metrics SG is scraped because it has metrics to scrape, and the events SG writes directly to ElasticSearch.

Although you bring up a very good point about scraping the SGs themselves for additional information. I suppose I'm not sure that the SG itself provides any additional information beyond what it is exposing, and in the case of the events SG, I'm not sure what metrics itself exposes.

Definitely something to consider, and I'd prefer to try and get as much of this correct the first time around as possible so that we reduce our release burdens.